### ISSUE TITLE
Error: Cannot find module 'https-proxy-agent'

### ISSUE THREAD

**Issue by dklinger** [Apollon77/ioBroker.alexa2#499]
Hi, 
after upgrading to the latest version, I get the error "Error: Cannot find module 'https-proxy-agent'" and the adapter isn't starting.
I'm on js-controller 3.1.4.
I was on node-js 10 and already upgraded to 12.
I also did a npm rebuild already.

Any ideas what the problem might be?

host.raspiiobroker | 2020-06-30 18:16:32.116 | error | instance system.adapter.alexa2.0 terminated with code 1 (JS_CONTROLLER_STOPPED)
-- | -- | -- | --
host.raspiiobroker | 2020-06-30 18:16:32.116 | error | Caught by controller[0]: }
host.raspiiobroker | 2020-06-30 18:16:32.116 | error | Caught by controller[0]: ]
host.raspiiobroker | 2020-06-30 18:16:32.115 | error | Caught by controller[0]: '/opt/iobroker/node_modules/iobroker.alexa2/main.js'
host.raspiiobroker | 2020-06-30 18:16:32.115 | error | Caught by controller[0]: '/opt/iobroker/node_modules/virtual-device-sdk/lib/src/Index.js',
host.raspiiobroker | 2020-06-30 18:16:32.114 | error | Caught by controller[0]: '/opt/iobroker/node_modules/virtual-device-sdk/lib/src/VirtualDevice.js',
host.raspiiobroker | 2020-06-30 18:16:32.114 | error | Caught by controller[0]: requireStack: [
host.raspiiobroker | 2020-06-30 18:16:32.113 | error | Caught by controller[0]: code: 'MODULE_NOT_FOUND',
host.raspiiobroker | 2020-06-30 18:16:32.113 | error | Caught by controller[0]: at Module.require (internal/modules/cjs/loader.js:1026:19) {
host.raspiiobroker | 2020-06-30 18:16:32.112 | error | Caught by controller[0]: at Function.Module._load (internal/modules/cjs/loader.js:879:14)
host.raspiiobroker | 2020-06-30 18:16:32.112 | error | Caught by controller[0]: at Module.load (internal/modules/cjs/loader.js:986:32)
host.raspiiobroker | 2020-06-30 18:16:32.111 | error | Caught by controller[0]: at Object.Module._extensions..js (internal/modules/cjs/loader.js:1158:10)
host.raspiiobroker | 2020-06-30 18:16:32.111 | error | Caught by controller[0]: at Module._compile (internal/modules/cjs/loader.js:1138:30)
host.raspiiobroker | 2020-06-30 18:16:32.110 | error | Caught by controller[0]: at Object.<anonymous> (/opt/iobroker/node_modules/virtual-device-sdk/lib/src/VirtualDevice.js:41:23)
host.raspiiobroker | 2020-06-30 18:16:32.110 | error | Caught by controller[0]: at require (internal/modules/cjs/helpers.js:72:18)
host.raspiiobroker | 2020-06-30 18:16:32.109 | error | Caught by controller[0]: at Module.require (internal/modules/cjs/loader.js:1026:19)
host.raspiiobroker | 2020-06-30 18:16:32.109 | error | Caught by controller[0]: at Function.Module._load (internal/modules/cjs/loader.js:842:27)
host.raspiiobroker | 2020-06-30 18:16:32.109 | error | Caught by controller[0]: at Function.Module._resolveFilename (internal/modules/cjs/loader.js:966:15)
host.raspiiobroker | 2020-06-30 18:16:32.108 | error | Caught by controller[0]: - /opt/iobroker/node_modules/iobroker.alexa2/main.js
host.raspiiobroker | 2020-06-30 18:16:32.108 | error | Caught by controller[0]: - /opt/iobroker/node_modules/virtual-device-sdk/lib/src/Index.js
host.raspiiobroker | 2020-06-30 18:16:32.107 | error | Caught by controller[0]: - /opt/iobroker/node_modules/virtual-device-sdk/lib/src/VirtualDevice.js
host.raspiiobroker | 2020-06-30 18:16:32.107 | error | Caught by controller[0]: Require stack:
host.raspiiobroker | 2020-06-30 18:16:32.106 | error | Caught by controller[0]: Error: Cannot find module 'https-proxy-agent'
host.raspiiobroker | 2020-06-30 18:16:32.106 | error | Caught by controller[0]: ^
host.raspiiobroker | 2020-06-30 18:16:32.105 | error | Caught by controller[0]: throw err;
host.raspiiobroker | 2020-06-30 18:16:32.104 | error | Caught by controller[0]: internal/modules/cjs/loader.js:969
host.raspiiobroker | 2020-06-30 18:16:30.841 | info | instance system.adapter.alexa2.0 started with pid 16268

**Comment by dklinger** [comment]
I also removed everything of the adapter, tried the installation fixer and did a downgrade to the previous version.
Still without success. I'm out of ideas.

**Comment by Apollon77** [comment]
npm somehow messes up with somepackages. tryo to run

npm install iobroker.alexa2 --production 

in your iobroker directory

**Comment by dklinger** [comment]
Unfortunately this also didn't help.
But upgrading to the newest js-controller and also running the reinstall-script did the trick.
Thanks.

**Comment by Lukas0802** [comment]
@dklinger what reinstall-script do you mean?
Unfortunately I get the same error messages since the npm update. Reinstallation didn't help either, nor did using the old version with the old data (on Docker).

**Comment by dklinger** [comment]
Hi @Lukas0802,
I used the reinstall.sh in the data folder.
I don't know what the Problem was but that worked for me.

**Comment by Lukas0802** [comment]
Ok, thank you!
Unfortunately, that didn't work for me :( I guess, I have to reinstall my complete ioBroker instance...

**Comment by Apollon77** [comment]
No just reinstall Alexa2

In iobroker dir

npm i iobroker.alexa2 --production

**Comment by Lukas0802** [comment]
I also tested a reinstallation of Alexa2, but it did not help either.

**Comment by Apollon77** [comment]
And the error is the same?? Module not found? Then our npm stuff seems broken somehow. 

You can check the node_module folders of the affected dependency folder and delete manuall and then try to reinstall again

**Comment by Lukas0802** [comment]
Yes.
I tried the following and the error message was always the same ("cannot find module https-proxy-agent"):
- Reinstallation of Alexa2
- Executing the reinstall.js script
- npm rebuild (because of the zigbee adapter)
- iobroker upgrade self
- Downgrade to a Docker Container with older Node.js version (10) and my old data

The steps were also all performed independently.

I have now reinstalled the complete iobroker instance and everything works again.
